ls1intum/Artemis
# [EXAM] Quiz exercise management

### Checklist

### Motivation and Context
Quiz exercise CRUD operations must be adapted for exam mode.

### Description
Client:
- [x] Created new client routes for quiz-exercise update and creation
For creation: ':examId/exercise-groups/:groupId/quiz-exercises/new'
For update: ':examId/exercise-groups/:groupId/quiz-exercises/:exerciseId/edit'
- [x] Load quiz exercise module
- [x] Deactivate categories in exam mode for now, as this relies on a Course
- [x] Added "Add quiz" button to exercise group component  

Server:
- [x] Deactivate Tutor/Student notification on exercise creation/update
- [x] Add additional authorization check to GET quiz exercise route. Only instructors and admins should be allowed to access an exam quiz exercise
- [x] Adapted authorization check for deletion

### Steps for Testing
_Ensure that the existing CRUD operations for regular quiz exercises still work_

1. Log in as admin or instructor
2. Create a regular quiz exercise
--> Exercise was created
3. Update the exercise
--> Exercise was updated
4. View the exercise
--> Can open the page
5. Delete the exercise
--> Exercise was deleted

_Ensure that CRUD operations for exam quiz exercises work_

1. Log in as admin or instructor
2. Create an exam
3. Create an exercise group
4. Create a quiz exercise
--> Exercise was created
5. Edit the quiz exercise
--> Update was saved
6. Navigate to `/course-management/:courseId/quiz-exercises/:exerciseId` using the id of an exam quiz exercise
--> Should be able to see the exercise
7. Delete the exercise
--> Exercise was deleted

_Ensure correct authorization for quiz exercises_
1. Log in as a tutor or student
2. Navigate to `course-management/:courseId/exams/:examId/exercise-groups/:groupId/quiz-exercises/new`
--> You should not be able to access the route
4. Use the id of an exam exercise for the following steps
5. Navigate to `course-management/:courseId/exams/:examId/exercise-groups/:groupId/quiz-exercises/:exerciseId/edit`
--> You should not be able to access the route
7. Navigate to `/course-management/:courseId/quiz-exercises/:exerciseId`
--> You should not be able to see the exercise

### Screenshots
